src/ngraph/provenance_config.cpp

```diff
@@ -0,0 +1,25 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/provenance_config.hpp"
+#include <cstdlib>
+
+using namespace ngraph;
+
+bool ProvenanceConfig::is_enabled() const
+{
+    return std::getenv("NGRAPH_PROVENANCE_ENABLE");
```

Comment:
make a static variable inside this function and store
```
bool s_provenanace_enabled = std::getenv("NGRAPH_PROVENANCE_ENABLE") != nullptr;
```
then return s_provenance_enabled